src/MongoDB/Driver/ReadPreference.cpp

```diff
@@ -82,16 +82,18 @@ int64_t HHVM_METHOD(MongoDBDriverReadPreference, getMode)
 {
 	MongoDBDriverReadPreferenceData* data = Native::data<MongoDBDriverReadPreferenceData>(this_);
 
-	return data->m_read_preference->mode;
+	return mongoc_read_prefs_get_mode(data->m_read_preference);
 }
 
 Array HHVM_METHOD(MongoDBDriverReadPreference, getTagSets)
 {
 	MongoDBDriverReadPreferenceData* data = Native::data<MongoDBDriverReadPreferenceData>(this_);
 	Variant v_tags;
+	const bson_t *tags;
 	
 	hippo_bson_conversion_options_t options = HIPPO_TYPEMAP_DEBUG_INITIALIZER;
-	BsonToVariantConverter convertor(bson_get_data(&data->m_read_preference->tags), data->m_read_preference->tags.len, options);
+	tags = mongoc_read_prefs_get_tags(data->m_read_preference);
```

Comment:
Any reason this isn't consolidated into the declaration, as you did in `MongoDBDriverCursor::__debugInfo`?